### DISCUSSION TITLE
Update gps_position while driving

### DISCUSSION THREAD

**Discussion by FilipDem** [bimmerconnected/bimmer_connected#420] [Feature request]
Hey,
I would like to implement a feature in my Domoticz plugin based on the geolocation. When arriving nearby home, I would trigger an action... Therefore I would need the combination of driving (`is_vehicle_active`) and the GPS position.
However I see that this is explicitly blocked and no GPS position is returned while driving.

I am wondering whether this is due to privacy reason in your implementation or rather a restriction of the BMW remote services that don't allow to get back the GPS position while driving.

Thanks for feedback

```
    def gps_position(self) -> Tuple[float, float]:
        """Get the last known position of the vehicle.
        Returns a tuple of (latitude, longitude).
        This only provides data, if the vehicle tracking is enabled!
        """
        if self.is_vehicle_active:
            _LOGGER.info('Vehicle was moving at last update, no position available')
            return (None, None)
        if not self._remote_service_position and "vehicleLocation" not in self.properties:
            _LOGGER.info("No vehicle location data available.")
            return (None, None)

```

**Comment by sincze** [comment]
Hi [FilipDem](https://github.com/FilipDem) 
I've been using a different implementation of Connected-Drive based on PHP and the GPS data is provided every 30 sec or so. I plot these on a map. The data should be somewhere, but maybe can be added to this great project as well.

![image](https://user-images.githubusercontent.com/5776333/154650091-ebf70eac-8abd-4a23-9afb-2042d98069e7.png)

**Comment by rikroe** [comment]
Hi both, you should be able to use the [`0.9.0b1`](https://github.com/bimmerconnected/bimmer_connected/releases/tag/v0.9.0b1) to get this functionality.
I know that I removed the condition there. Please beware of the breaking changes and remember that this is a major rework and might still contain bugs. If you find any, please create an issue!

**Reply by sincze** [reply]
Works like a charm!!

**Comment by FilipDem** [comment]
Hi,
I just tried to update my Domoticz plugin with the breaking changes. I am using threading mechanism inside Domoticz as recommended for the Domoticz plugin development.
However when calling the `MyBMWAccount(USERNAME, PASSWORD, Regions.REST_OF_WORLD)` inside the thread, I receive the error `There is no current event loop inside the thread`. I suppose it has something to do with the asyncio...
Any hint how to get this solved? 
See [https://github.com/FilipDem/Domoticz-BMW-plugin/blob/main/plugin.py](https://github.com/FilipDem/Domoticz-BMW-plugin/blob/main/plugin.py), function `handleTasks` (line 210).
Thanks,
F

**Comment by S54Mpower** [comment]
As far as I am aware, GPS location output is blocked by BMW servers when vehicle in motion.

**Reply by rikroe** [reply]
It heavily depends on the car, some vehicles also report their location to the API while driving or allow the find vehicle service while driving.

Some don't.
